flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTracker.java

```diff
@@ -39,7 +39,8 @@ void startTrackingPartition(
             ResultPartitionDeploymentDescriptor resultPartitionDeploymentDescriptor);
 
     /** Releases the given partitions and stop the tracking of partitions that were released. */
-    void stopTrackingAndReleasePartitions(Collection<ResultPartitionID> resultPartitionIds);
+    void stopTrackingAndReleasePartitions(
```

Comment:
Maybe keep the old method and add this as a new one, with default implementation that `releaseOnShuffleMaster` is true?
I think the second param is not easy for understanding so I prefer to not force set it unless necessary.